File: C++/day9Recursion.cpp
#include <iostream>
#include <cstdlib>

using namespace std;

int min(int x, int y) {
  int outcome = x;
  if (y <= x) { outcome = y; }

  return outcome;
}

int max(int x, int y) {
  int outcome = x;
  
  if (y >= x) { outcome = y; }
  
  return outcome;
}

int gcd(int x, int y) {
  int outcome = y;
  if (x == y) {
    outcome = x;
  } else {
    outcome = gcd(max(x, y) - min(x, y), min(x, y));
  }

  return outcome;
}

int main() {

  int x, y = 0;
  cin >> x >> y;
  cout << gcd(x,y) << endl;

  return(0);
}

File: SWIFT/day9Recursion.swift
import Foundation


func min(x x: Int, y: Int) -> Int {
  var outcome = x
  
  if y <= x {
    outcome = y
  }
  return outcome
}

func max(x x: Int, y: Int) -> Int {
  var outcome = x
  
  if y >= x {
    outcome = y
  }
  
  return outcome
}
func gcd(x x: Int, y: Int) -> Int {
  if x == y {
    return x
  } else {// if x > y {
    return gcd(x: max(x: x, y: y) - min(x: x, y: y), y: min(x: x, y: y))
  }
}

let aI = readLine()!.characters.split(" ").map { Int(String($0))! }
print(gcd(x: aI[0], y: aI[1]))

